frontend/src/js/controllers/challengeCtrl.js

```diff
@@ -1415,6 +1415,88 @@
             }
         };
 
+        vm.challengePhaseDialog = function(ev, phase) {
+            vm.page.challenge_phase = phase;
+            vm.phase_start_date = new Date(phase.start_date);
+            vm.phase_end_date = new Date(phase.end_date);
+            vm.test_annotation_file = null;
+            vm.sanity_check_pass = true;
+            vm.sanity_check = "";
+            $mdDialog.show({
+                scope: $scope,
+                preserveScope: true,
+                targetEvent: ev,
+                templateUrl: 'dist/views/web/challenge/edit-challenge/edit-challenge-phase.html',
+                escapeToClose: false
+            });
+        };
+
+        vm.sanityCheckInit = function() {
+            vm.sanity_check_pass = false;
+        };
+
+        vm.sanityCheckPass = function() {
+            vm.sanity_check_pass = true;
+        };
+
+        vm.editChallengePhase = function(editChallengePhaseForm) {
+            if(editChallengePhaseForm){
+                vm.page.challenge_phase.start_date = vm.phase_start_date;
+                vm.page.challenge_phase.end_date = vm.phase_end_date;
+                vm.challengePhaseId = vm.page.challenge_phase.id;
+                var parameters = {};
+                parameters.url = "challenges/challenge/" + vm.challengeId + "/challenge_phase/" + vm.challengePhaseId;
+                parameters.method = 'PATCH';
+                parameters.data = vm.page.challenge_phase;
+                if (vm.test_annotation_file){
+                    var formData = new FormData();
+                    formData.append("test_annotation", vm.test_annotation_file);
+                    parameters.data.test_annotation = formData;
+                }
+                parameters.token = userKey;
+                parameters.headers = {'Content-Type': 'application/x-www-form-urlencoded'};
```

Comment:
I think there is no need to add the header specifically. Please check `services.js`, we have a service for it.

Reply:
@RishabhJain2018 Is it `sendRequest`? How do I use it?